### PR TITLE
Update to SpiderMonkey 112

### PR DESCRIPTION
**Pull request by JakeChampion**
I thought I did the 12,000+ commit rebase successfully of SpiderMonkey 110 to 112 (with our patches)...

It looks like perhaps I was not successful as we have some new test failures for the streams implementation

### CODE REVIEW

**Comment by guybedford** [comment]
It seems like the issue here is for:

```js
Object.defineProperty(Object.prototype, 'start', get() { return 'start' });
```

which is throwing:

```
Unhandled rejection with value: object "TypeError: can't define property "start": Object is not extensible"
```

So somehow, the global intrinsics are being frozen, which sounds like `setFreezeBuiltins` somehow changed.

I tried adding to `js-compute-runtime.cpp`:

```
JS::RealmOptions realm_options;
realm_options.creationOptions().setFreezeBuiltins(false);
```

But that doesn't seem to resolve the issue. Seems like we should figure this out...

File: c-dependencies/js-compute-runtime/js-compute-runtime.cpp [hunk 1]
```diff
@@ -117,13 +117,8 @@ bool init_js() {
 
   // TODO: check if we should set a different creation zone.
   JS::RealmOptions options;
-  options.creationOptions()
-      .setStreamsEnabled(true)
-      .setReadableByteStreamsEnabled(true)
-      .setBYOBStreamReadersEnabled(true)
-      .setReadableStreamPipeToEnabled(true)
-      .setWritableStreamsEnabled(true)
```
Comment (JakeChampion) [inline c-dependencies/js-compute-runtime/js-compute-runtime.cpp:125]: These have all been merged into the streamsEnabled option 👍

File: c-dependencies/js-compute-runtime/builtins/transform-stream.cpp [hunk 2]
```diff
@@ -222,7 +224,7 @@ bool ExtractStrategy(JSContext *cx, JS::HandleValue strategy, double default_hwm
     if (!JS::ToNumber(cx, val, hwm)) {
       return false;
     }
-    if (mozilla::IsNaN(*hwm) || *hwm < 0) {
+    if (std::isnan(*hwm) || *hwm < 0) {
```
Comment (JakeChampion) [inline c-dependencies/js-compute-runtime/builtins/transform-stream.cpp:227]: Between Mozilla Gecko 110 and 112, they removed mozilla::IsNaN - I've replaced the call with its `std` equivalent, which is also what is now used in Gecko's codebase

**Comment by JakeChampion** [comment]
@guybedford and @elliottt -- The upgrade is now fully working, all the same tests pass as before the upgrade ☺️

**Review by elliottt — APPROVED** [review]
Awesome! :tada:

**Pull request merged**